mono/mini/mini-exceptions.c

```diff
@@ -3095,6 +3107,53 @@ mono_llvm_match_exception (MonoJitInfo *jinfo, guint32 region_start, guint32 reg
 		}
 	}
 
+ // If no trace_ips, then a rethrow
+	if (index != -1 && ((MonoException *)exc)->trace_ips == NULL) {
+#ifdef MONO_ARCH_HAVE_UNWIND_BACKTRACE
+		// Find the number of frames above us
+		size_t count = 0;
+		_Unwind_Backtrace (count_frames, &count);
+
+		GList *forward = jit_tls->trace_ips;
+
+		// Skip the frames above us
+		for (size_t i = 1; i < count; i++)
+			forward = forward->next;
+
+		g_assert (forward != NULL);
+
+		// Build stacktrace backwards, move cursor to end
+		GList *backward = forward;
+		int size = 1;
+		while(backward->next) {
+			backward = backward->next;
+			size++;
+		}
+		/* The list contains ip-gshared info pairs */
+		size = size * 2;
```

Comment:
This sorta changes the meaning of `size` from “number of pairs” to “number of elements”. Either introduce a different variable or use consistent counting throughout.